base/abstractarray.jl

```diff
@@ -761,8 +781,9 @@ function cat_t(catdims, typeC::Type, X...)
     for i = 2:nargs
         for d = 1:ndimsC
             currentdim = (d <= ndimsX[i] ? size(X[i],d) : 1)
-            if dims2cat[d]==0
-                dimsC[d] == currentdim || throw(DimensionMismatch("mismatch in dimension $(d)"))
+            if dims2cat[d] == 0
+                dimsC[d] == currentdim ||
+                    throw(DimensionMismatch("mismatch in dimension $(d) (expected $currentdim, got $(dimsC[d]))"))
```

Comment:
I think I'd write this the other way around: `(expected $(dimsC[d]), got $currentdim)`.  The expected value is your previous array, not the current one.